From 3e497d5fd5f828dfad950408c2e86f5cbc37728d Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Tue, 30 Apr 2013 14:11:02 +0800
Subject: MPEG4Extractor: Avoid some unexpected rounded situation when calculating sample time

BZ: 48862

Original patches:
R3: 30130 (BZ: 18544)
R1: 7415 (BZ: 739)

Root cause: Normally the sampletime can be divided without remainder,
a correct frame can be select as the thumbnail. But if remainder exists
in unexpected situation and the sample time will be rounded, the calculated
thumbnail sample time will smaller than the orignail and the framework will
select the previous key frame other then the correct one.

Category: aosp improvement
Domain: Video.Media-container
Origin: internal
Upstream-Candidate: yes

Orig-Change-Id: Icfd9400f64bf584d8cf8d33fed61431402d2ee28
Change-Id: Ie381c82f01498880a0ddf4da5b23dedaba3d1c3c
Signed-off-by: Weian Chen <weian.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
---
 media/libstagefright/MPEG4Extractor.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index e93cb53..94fb356 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -450,7 +450,7 @@ sp<MetaData> MPEG4Extractor::getTrackMetaData(
                             &sampleTime) == OK) {
                     track->meta->setInt64(
                             kKeyThumbnailTime,
-                            ((int64_t)sampleTime * 1000000) / track->timescale);
+                            ((((int64_t)sampleTime * 10000000) / track->timescale) + 9) / 10);
                 }
             }
         }
-- 
1.7.4.1

